src/core/qgsproject.h

```diff
@@ -887,7 +887,14 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
      */
     void layerLoaded( int i, int n );
 
-    void loadingLayer( const QString & );
+    void loadingLayer( const QString &layerName );
+
+    /**
+     * @brief Emitted when loading layers has produced some messages
+     * @param layerName the layer name
+     * @param messages a list of pairs of Qgis::MessageLevel and messages
+     */
+    void loadingLayerMessages( const QString &layerName, const QList<QPair<Qgis::MessageLevel, QString>> &messages );
```

Comment:
This doesn't sound like a signal to me - more like a getter. Maybe loadingLayerMessageReceived is better?

Also, you broke your own lovely pyqgis docs by using "@" notation!

Is this a since 3.0 or 3.2?

Reply:
fixed